Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/ui/details/TestCertificateDetailsFragment.kt

```diff
@@ -259,15 +250,17 @@ class TestCertificateDetailsFragment : Fragment(R.layout.fragment_test_certifica
     }
 
     private fun showCertificateDeletionRequest() {
```

Comment:
⬇️ 
This should not be part of a Fragment..
We already have a DccValidityStateChangeObserver. You can use that and update the notification accordingly if needed

Reply:
Yes, but I don't want to update the notification when the state of the DCC changes. I might have seen the notification already and dismissed it, and then the state changes and I could trigger the notification again. I just want to check if the notification is already emitted, and if it is, then update it when I'm deleting the DCC so it will no longer point to the certificate details screen.

Why shouldn't this be part of a fragment?

Reply:
I just don't get what is wrong with the previous approach? We always look for the certificate by container ID and  if it is not there or recycled simply move back.

Reply:
The "old" approach was [introduced](https://github.com/corona-warn-app/cwa-app-android/pull/5129) a few weeks ago. Then this bug [12991](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12991) showed up and I think the reason is because sometimes, after scanning a new cert, we get to the details screen too quickly, and the repo didn't update yet, so we called the `popBackstack()` function that lead us away from the screen. 

Why do we keep returning to this approach of Go to screen ->  check if anything can be displayed -> go to other screen if there's nothing to be displayed. We've already had a few bugs because of this, and even though the calculation happens fast, sometimes you can still see the screen for a fraction of a second before reaching the right screen. If we can determine the final destination before travelling to another screen, why try to jump through other screens?

Reply:
My idea was to use _DccValidityStateChangeObserver_ as a trigger for a function which handles the logic to check if a notification needs to be updated.
Putting such logic inside a Fragment has the downsides that it is hard to test and you have to repeat it multiple times, like we see in this PR. Also this code only gets executed if the Fragment is displayed. Are those the only situations where a certificate can get removed? What happens if a certificate gets replaced by reissuance?

Reply:
Alright, I understand now. This makes sense. I will try to do something like this to decouple the logic from the fragments.

Reply:
Gave up on trying to update the notification. The fix now is to simply create a separate flow with the certificates that will give us the latest value more quickly in the details screen.